Args.cs

```diff
@@ -12,80 +18,136 @@ public class Args
-                    {
-                        this.VstsPat = args[i + 1];
-                        ++i;
-                        continue;
+                    if (this.IsValidArg(args, i) && !this.IsValidArg(args, i + 1))
+                    {
+                        this.VstsPat = args[i + 1];
+                        ++i;
+                        continue;
                     }
                 }
 
                 if (this.ArgEquals(args[i], "DropDestination", "d"))
                 {
-                    if (this.IsValidArg(args, i) && !this.IsValidArg(args, i + 1))
-                    {
-                        this.DropDestination = args[i + 1];
-                        ++i;
-                        continue;
+                    if (this.IsValidArg(args, i) && !this.IsValidArg(args, i + 1))
+                    {
+                        this.DropDestination = args[i + 1];
+                        ++i;
+                        continue;
                     }
                 }
 
                 if (this.ArgEquals(args[i], "DropUrl", "u"))
                 {
-                    if (this.IsValidArg(args, i) && !this.IsValidArg(args, i + 1))
-                    {
-                        this.DropUrl = args[i + 1];
-                        ++i;
-                        continue;
+                    if (this.IsValidArg(args, i) && !this.IsValidArg(args, i + 1))
+                    {
+                        this.DropUrl = args[i + 1];
+                        ++i;
+                        continue;
                     }
                 }
 
                 if (this.ArgEquals(args[i], "RelativePath", "p"))
                 {
-                    if (this.IsValidArg(args, i) && !this.IsValidArg(args, i + 1))
-                    {
-                        this.RelativePath = args[i + 1];
-                        ++i;
-                        continue;
+                    if (this.IsValidArg(args, i) && !this.IsValidArg(args, i + 1))
+                    {
+                        this.RelativePath = args[i + 1];
+                        ++i;
+                        continue;
                     }
                 }
+
             }
+
+            this.CheckEnvironmentVariables();
+            this.ValidatePat();
         }
 
-        private bool ArgEquals(string givenArgument, string expected, string shorthandArg = null)
-        {
-            char[] trimChars = { '-' };
-            expected = expected.Trim(trimChars);
-            shorthandArg = shorthandArg.Trim(trimChars);
-
-            bool shortEq = false;
-            if (!string.IsNullOrWhiteSpace(shorthandArg))
-            {
-                shortEq = givenArgument.Equals(string.Format("-{0}", shorthandArg));
-            }
-
-            return shortEq || givenArgument.Equals(string.Format("--{0}", expected), StringComparison.OrdinalIgnoreCase);
+        public void CheckEnvironmentVariables()

```

Comment:
so you could not do the arg checks if you instead make this a static Args FromEnvVars() {} method and use it to create your args.

Reply:
then remove it from parse